node/podcontroller.go

```diff
@@ -228,6 +218,11 @@ func NewPodController(cfg PodControllerConfig) (*PodController, error) {
 		podStatusQ:         workqueue.NewNamedRateLimitingQueue(cfg.RateLimiter, "syncPodStatusFromProvider"),
 		podEventFilterFunc: cfg.PodEventFilterFunc,
 	}
+	if cfg.ConfigMapInformer != nil && cfg.SecretInformer != nil && cfg.ServiceInformer != nil {
+		pc.configMapLister = cfg.ConfigMapInformer.Lister()
```

Comment:
My suggestion is that you make these encapsulated in some structure like listers, and then you do something like:

```go

func (l *listers) initialized() bool {
   return l != nil
}
```